tests/test_insert_deletes_consolidate.cpp

```diff
@@ -159,209 +211,114 @@ void build_incremental_index(
-
-  if (points_to_delete_from_beginning > max_points_to_insert) {
-    points_to_delete_from_beginning =
-        static_cast<unsigned>(max_points_to_insert);
-    std::cerr << "WARNING: Reducing points to delete from beginning to "
-              << points_to_delete_from_beginning
-              << " points since the data file has only that many" << std::endl;
-  }
-
-  if (concurrent) {
-    int               sub_threads = (thread_count + 1) / 2;
-    bool              delete_launched = false;
-    std::future<void> delete_task;
-
-    diskann::Timer timer;
-
-    for (size_t start = current_point_offset; start < last_point_threshold;
-         start += points_per_checkpoint,
-                current_point_offset += points_per_checkpoint) {
-      const size_t end =
-          std::min(start + points_per_checkpoint, last_point_threshold);
-      std::cout << std::endl
-                << "Inserting from " << start << " to " << end << std::endl;
-
-      auto insert_task = std::async(std::launch::async, [&]() {
-        load_aligned_bin_part(data_path, data, start, end - start);
-        insert_till_next_checkpoint(index, start, end, sub_threads, data,
-                                    aligned_dim);
-      });
-      insert_task.wait();
-
-      if (!delete_launched && end >= start_deletes_after &&
-          end >= points_to_skip + points_to_delete_from_beginning) {
-        delete_launched = true;
-        params.Set<unsigned>("num_threads", sub_threads);
-
-        delete_task = std::async(std::launch::async, [&]() {
-          delete_from_beginning(index, params, points_to_skip,
-                                points_to_delete_from_beginning);
-        });
-      }
+  std::cout << "Initial non-incremental index build time for " << build_point
+            << " points took " << elapsedSeconds << " seconds ("
+            << build_point / elapsedSeconds << " points/second)\n ";
+
+  int64_t insert_index_size = max_points * insert_percentage;
+  if (insert_percentage > 0) {
+    if (delete_percentage > 0) {
+      load_aligned_bin_part(data_path, data, points_to_skip, build_point);
+    } else {
+      diskann::alloc_aligned(
+          (void**) &data,
+          (build_point + insert_index_size) * aligned_dim * sizeof(T),
+          8 * sizeof(T));
+      std::vector<TagT> tags(build_point + max_points * insert_percentage);
+      std::iota(tags.begin(), tags.end(), static_cast<TagT>(points_to_skip));
+      load_aligned_bin_part(data_path, data, points_to_skip,
+                            (build_point + insert_index_size));
     }
-    delete_task.wait();
-
-    std::cout << "Time Elapsed " << timer.elapsed() / 1000 << "ms\n";
-    const auto save_path_inc = get_save_filename(
-        save_path + ".after-concurrent-delete-", points_to_skip,
-        points_to_delete_from_beginning, last_point_threshold);
-    index.save(save_path_inc.c_str(), true);
-
+    insert_till_next_checkpoint(index, max_points, insert_percentage,
```

Comment:
Looks it may cause insert keys that already in the table (insert range is points_to_skip to build_point + insert_index_size)

Reply:
My random insertion scheme is that when there are insert and delete, it can be randomly selected from the complete set. For only insert, since the full amount needs to be kept with the build, if I randomly take the data set during build, then the data set when I insert is equivalent to random (for example, the full amount is 100,000, and the front of the random data set is taken during build. 70,000, and insert takes the last 30,000 data sets), so no insert the points are repeated